File: engine/flagship/impl/src/main/scala/jellon/ssg/engine/flagship/hints/FlagshipHintHandlers.scala
package jellon.ssg.engine.flagship.hints

import jellon.ssg.io.impl.HintHandlers
import jellon.ssg.io.spi.IOutputStreamResources
import org.springframework.stereotype.Component

@Component
class FlagshipHintHandlers(resources: IOutputStreamResources) extends HintHandlers(Seq(
  new JavaHintHandler(resources)
)) {

}

File: engine/flagship/guice/src/main/java/jellon/ssg/engine/flagship/integ/FlagshipHintHandlers.java
package jellon.ssg.engine.flagship.integ;

import jellon.ssg.io.spi.IOutputStreamResources;

import javax.inject.Inject;

/**
 * A Guice-Scala integration work-around. There doesn't appear to be a way to annotate an idiomatic scala constructor
 * with Inject.
 */
public class FlagshipHintHandlers extends jellon.ssg.engine.flagship.hints.FlagshipHintHandlers {
    @Inject
    public FlagshipHintHandlers(IOutputStreamResources resources) {
        super(resources);
    }
}
